example/constructorinjection/app.d

```diff
@@ -57,7 +64,10 @@ void main() {
 	auto dependencies = new shared DependencyContainer();
 	dependencies.register!Scheduler;
 	dependencies.register!Calendar;
-	dependencies.register!HardwareClock;
+	dependencies.register!HardwareClock( {
+		writeln("Running the creator");
+		return new HardwareClock("clock name");
+	});

```

Comment:
Perhaps create a separate example for this. This example is focused on injection via constructors, not creation.